winterfell/services/userService.js

```diff
@@ -77,3 +79,74 @@ module.exports.upsertUserValues = function(userId, keyValues, callback) {
     userValue.save(callback);
   });
 };
+
+/**
+ * Expected input object from idme:
+ * {
+ *  id
+ *  verified
+ *  affiliation
+ *  fname
+ *  lname
+ *  zip
+ *  uuid
+ *  email
+ * }
+ */
+module.exports.createNewUserFromIdMe = function(idmeInfo, callback) {
+  var newUser = {
+    firstname: idmeInfo.fname,
+    middlename: null,
+    lastname: idmeInfo.lname,
+    email: idmeInfo.email,
+    password: "fakepassword", // todo: generate a short random 8-char password
+  };
+  User.quickCreate(newUser, function(err, user) {
+    // update user's zip code?
+    callback(err, user);
+  });
+};
+
+// Method to add an ACTIVE socialUser to a User
+module.exports.addSocialUser = function(userId, type, token, callback) {
+  var now = Date.now();
+  User.findOne({_id: userId}, function(err, user) {
+    user.socialUsers.push({
+      type: type,
+      oauthToken: token,
```

Comment:
I'm pretty sure. Normally you use that token to make future queries with social providers. I haven't fully looked at what you could with Id.me but I know it's a thing for facebook.